src/bundle/settings.rs

```diff
@@ -76,22 +83,14 @@ impl Settings {
             Some(triple) => Some((triple.to_string(), TargetInfo::from_str(triple)?)),
             None => None,
         };
-        let target_dir = {
-            let mut path = current_dir.join("target");
-            if let Some((ref triple, _)) = target {
-                path.push(triple);
-            }
-            path.push(if is_release { "release" } else { "debug" });
-            path
-        };
-        let cargo_settings: CargoSettings = {
-            let toml_path = current_dir.join("Cargo.toml");
-            let mut toml_str = String::new();
-            let mut toml_file = File::open(toml_path)?;
-            toml_file.read_to_string(&mut toml_str)?;
-            toml::from_str(&toml_str)?
+        let cargo_settings = Settings::cargo_settings(&current_dir)?;
+        let package = match cargo_settings.package {
+            Some(package_info) => package_info,
+            None => return Err("No 'package' info found in 'Cargo.toml'".into())
```

Comment:
In this case, this can be written more succinctly as just:
```
None => bail!("No 'package' info found in 'Cargo.toml'"),
```

Reply:
I was getting some "note: this error originates in a macro outside of the current crate
" errors, but will try to fix them.